src/snowflake/snowpark/dataframe.py

```diff
@@ -2183,6 +2183,13 @@ def sort(
         """
 
         is_order_by_all = not cols
+
+        # This code performs additional type checks, run first.
+        exprs = self._convert_cols_to_exprs("sort()", *cols)
+        if not is_order_by_all:
+            if not exprs:
+                raise ValueError("sort() needs at least one sort expression.")
```

Comment:
do we need these three lines at all?
if is_order_by_all, it won't run
if not order_by_all, there must be exprs.